### ISSUE TITLE
Talk about hosting providers!

### ISSUE THREAD

**Issue by pde** [certbot/website#158]
I think we planned to have some cases in the instruction generator that talked about the fact that not everyone can use an ACME client -- some people need integration from their hosting provider.

But we don't seem to have done that...

**Comment by pde** [comment]
@SwartzCr I think this is important; can you figure out and shepherd some version of this while I'm away?

**Comment by SwartzCr** [comment]
Will do

**Comment by pconrad-fb** [comment]
What's the best way for me to get information to get started on this one?

**Comment by pde** [comment]
@pconrad-fb Depends on whether you're comfortable doing a little bit of JavaScript coding. If yes, I'd say clone the https://github.com/certbot/website repository, and use `grep -r` to follow the threads for how some of the existing instructions are generated.

If not, you should probably pair up with @SwartzCr to get a walkthrough of how to do things.

**Comment by pconrad-fb** [comment]
Sure, Javascript is fine. But how do I find out the information I'm actually documenting, about integration vs. ACME?

**Comment by SwartzCr** [comment]
So this will be a drop down item on the certbot website - I believe I wrote you a note about how to create new ones of those in a previous thread.
Info about which hosting providers support let's encrypt can be found here: https://community.letsencrypt.org/t/web-hosting-who-support-lets-encrypt/6920
and any of those (source) links should give you a sense of what they're like.
To add a new drop down item you'll want to add it to https://github.com/certbot/website/blob/master/_scripts/instruction-widget/data/inputs.json
so that it appears in the drop down, and then you'll have either instructions.js or install.js here https://github.com/certbot/website/tree/master/_scripts/instruction-widget point to a template here https://github.com/certbot/website/tree/master/_scripts/instruction-widget/templates

Let me know if you have any questions

**Comment by SwartzCr** [comment]
@pconrad-fb if you have questions about editing templates - can you provide feedback on this document as well? https://github.com/certbot/website/blob/master/EDITING.md
We'll be linking to it from certbot.eff.org and want to make sure that it's clear to new contributors

**Comment by SwartzCr** [comment]
@pconrad-fb have you had a chance to think about this more? I think the community.letsencrypt.org link should be sufficient to get you started, if not let me know and I can find more resources for you.
I know that we want this to be addressed 'soon' so that we can improve the first run experience for new users before we do a big publicity push around certbot, so I'm going to tag @schoen and @bmw so that they can weigh in on this too and keep an eye on it while I'm out of the office

**Comment by pconrad-fb** [comment]
I haven't, sorry, I've been swamped. I was hoping to look at it last Thursday (Thursdays are goo usually) and now I'm hoping to get a look at it tomorrow.

**Comment by pconrad-fb** [comment]
Looking at it today. Just to add a couple extra small obstacles, I've been issued a new laptop for work, which means I don't have my shell command history or my Apple Notes, so I'm having to reconstruct some of the steps. Stay tuned.

**Comment by pconrad-fb** [comment]
Okay, I'm back up and running. 

So. When someone selects a host from the third dropdown, what happens? How does it affect the docs downstream?

**Comment by bmw** [comment]
Great! Glad to have you back.

So after talking to everyone here about what we want, I think this issue is going to be a bit tricky. There's a fair amount we want here and we need this issue to be closed with the code live on the site in no more than three weeks. If you still want to take this issue, let me know, but otherwise it may make more sense to bounce this issue back to us and I can help you find another issue.

Let me know what you'd like to do.

**Comment by pconrad-fb** [comment]
Three weeks will be tricky what with Dreamforce coming up. Happy to help out with another issue that's a little less time-sensitive.

**Comment by bmw** [comment]
I think a good issue to continue working on is certbot/certbot#2938 back in the rst files in the Certbot repo. The problems described in the original post have been resolved, but our documentation which can be seen at https://certbot.eff.org/docs can continued to be cleaned up. When we met before, we talked about restructuring some of the sections such as the user guide or the developer guide. The documentation here is largely ad hoc and can certainly be cleaned up.

It might make sense for you to read through this documentation, see what we have and how we can improve it. I think there is a lot of benefit in having an outside set of eyes see what doesn't flow well, what doesn't make sense, etc. This is a fairly vague task, but this is something we talked about before and I think there is a lot of benefit in doing something like this. If you have questions or want to get our input, don't hesitate to ask. If you want, you can open a pull request against the Certbot repo and put "[WIP]" (which stands for "work in progress") in the title so we can comment on your work as you do it. If you're not interested in this or the task seems too vague, please let me know and I can help you find something else.

Thanks @pconrad-fb!

cc @pde

**Comment by pconrad-fb** [comment]
Cool. This is something I can dig in to. I'll go through this Thursday and put together a list of proposed actions.

**Comment by pconrad-fb** [comment]
Still going through it (Dreamforce impact has delayed me). Stay tuned.

**Comment by pconrad-fb** [comment]
I'd like to either 

a) Get rid of the Quick Install section, making sure that all the information about installation appears in Getting Certbot in an organized way, or

b) Change "Getting Certbot" to "Installing when no packages are available" and move the "Just go to the front page and choose your OS/webserver" instructions to Quick install, or

c) Change "Quick Install" to "Get Certbot" and remove "Getting Certbot" from the User Guide

I'm leaning heavily toward c) or a) in that order. @SwartzCr or anyone else have a preference?

Also, I'm in the process of fixing something that ended up being put out of place in the installation docs.

**Comment by bmw** [comment]
@pconrad-fb, sorry for the delay on the feedback. In the past week and a half, we've released the biggest Certbot update since our initial release and as a consequence have been scrambling to fix bugs.

I agree there is information duplicated between the Quick Install section and the User Guide. For example, [this](https://certbot.eff.org/docs/using.html#id1) is remarkably similar to the Quick Install section. To help inform our decision here, which of these three approaches do you prefer?

1. Completely separate installation instructions from the user guide.
2. Have basic installation instructions on one page and more in depth installation instructions in the user guide.
3. Have everything in the user guide.

I personally have a small preference for option 1 or 2 which makes me lean towards your option C as well. I will say though that I like the wording under Getting Certbot better as I think it more strongly pushes people towards https://certbot.eff.org and OS packages than the Quick Install section. Nearly all Certbot users should be using the advice given to them through the interactive instructions.

Related to this, I think our [other installation methods](https://certbot.eff.org/docs/using.html#other-installation-methods) section needs work. It appears it's a subcategory under commands and it duplicates a lot of information in the "Quick Install" guide as I linked above. Additionally, the OS instructions duplicate a lot of information found in the dropdowns at https://certbot.eff.org. This duplication makes it hard for us to maintain, however, if you think this information has value being both in the interactive instructions and our documentation, we might be able to solve the maintenance problem on our end through technical means.

In short, you certainly have the go ahead to do option C (perhaps with the caveat of taking some wording from the Getting Certbot section). I'm also interested in your thoughts at a higher level though and am inclined to trust your opinion on how a lot of these things should be structured. Let me know what you think and I look forward to seeing a pull request from you.

**Comment by pconrad-fb** [comment]
I like a separate "Get Certbot" section outside the User Guide, which would strongly encourage using the drop-down menus, followed by certbot-auto as a second-to-last resort, and finally the rest of the info. Let me try to take a look at this tomorrow or Thursday.

**Comment by bmw** [comment]
Sounds great. Thanks @pconrad-fb!

**Comment by bmw** [comment]
Closing this issue in favor of #137.

@pconrad-fb, if you'd like to continue this conversation, let's move it back to certbot/certbot#2938.
